### PR TITLE
Add sort_descending in the Python client Table class

### LINKED ISSUE #4083: sort_descending is not available from Python Client
**Description**

There is no sort_descending method on Table.

**Steps to reproduce**

session.run_script("from deephaven import empty_table\nx=empty_table(10).update([\"A=1\"])")
sorted = session.open_table("x").sort("A")

**Expected results**

The table is sorted descending by A.

**Actual results**

I get an error message saying sort_descending is not found.

**Versions**
 - Deephaven: 0.25.3

### PR DESCRIPTION
**Pull request by jmao-denver**
Fixes #4083

### CODE REVIEW

File: py/client/pydeephaven/table.py [hunk 1]
```diff
@@ -219,6 +220,23 @@ def sort(self, order_by: Union[str, List[str]], order: Union[SortDirection, List
         """
         return super().sort(order_by, order)
 
+    def sort_descending(self, order_by: Union[str, List[str]]) -> Table:
+        """The sort_descending method creates a new table where rows in a table are sorted in a largest to smallest
```
Comment (rcaudy) [inline py/client/pydeephaven/table.py:224]: Is largest to smallest the right way to phrase this? Descending order, or "last to first" or something might be better.

Comment (chipkent) [reply py/client/pydeephaven/table.py:224]:
```suggestion
        """The sort_descending method creates a new table where rows in a table are sorted in descending
```
I think this is most consistent with our `sort` docs.

File: py/client/pydeephaven/table.py [hunk 2]
```diff
@@ -219,6 +220,23 @@ def sort(self, order_by: Union[str, List[str]], order: Union[SortDirection, List
...
+            order_by (Union[str, List[str]]): the column name(s)
+
+        Returns:
+            a Table object
+
+        Raises:
+            DHError
+        """
+        order_by = to_list(order_by)
+        order = [SortDirection.DESCENDING] * len(order_by)
```
Comment (chipkent) [inline py/client/pydeephaven/table.py:237]:
```suggestion
        order = SortDirection.DESCENDING
```
I think this should be sufficient and is more simple.

File: py/client/pydeephaven/table.py [hunk 3]
```diff
@@ -219,6 +220,23 @@ def sort(self, order_by: Union[str, List[str]], order: Union[SortDirection, List
         """
         return super().sort(order_by, order)
 
+    def sort_descending(self, order_by: Union[str, List[str]]) -> Table:
```
Comment (chipkent) [inline py/client/pydeephaven/table.py:223]: * Should there be a `sort_ascending`?
* Is this needed if a user can simply do `sort(order_by, DESCENDING)`?
* If this method is here, should it also be on the proxy?

**Review by chipkent — CHANGES_REQUESTED** [review]

File: py/client/pydeephaven/table.py (hunk 3 above)
Comment (jmao-denver) [reply py/client/pydeephaven/table.py:223]: - It does seem odd to have it, but I think only someone who remembers its history can explain its existence and the need to keep it for backwards compatibility.
- see above
- not sure if we want to pollute the API when we have doubts about its very existence.

Comment (chipkent) [reply py/client/pydeephaven/table.py:223]: I see https://github.com/deephaven/deephaven-core/issues/4083.  The user wants the client API to match the server API, which is reasonable.  Looking at our server API, we only have `sort_descending`.  `sort_ascending` is accomplished via `sort` with default arguments.

Looking some more, the server has this method for:
* PartitionedTableProxy
* RollupNodeOperationsRecorder
* Table
* TreeNodeOperationsRecorder

Looking at the client, we just have Table, so it appears that this is the only needed change.

All that to say that I think the needed work is done.

**Review by chipkent — APPROVED** [review]

**Pull request merged**